packages/medusa/src/strategies/batch-jobs/product/export.ts

```diff
@@ -0,0 +1,420 @@
+                    region: price.region
+                      ? {
+                          currency_code: price.region.currency_code,
+                          name: price.region.name,
+                          id: price.region.id,
+                        }
+                      : null,
+                  })
+                )
+              })
+            }
+          }
+        }
+
+        offset += products.length
+        products = []
+      }
+
+      await this.batchJobService_
+        .withTransaction(transactionManager)
+        .update(batchJob, {
+          context: {
+            shape: {
+              dynamicImageColumnCount,
+              dynamicOptionColumnCount,
+              prices: [...pricesData].map((stringifyData) =>
+                JSON.parse(stringifyData)
+              ),
+            },
+          },
+          result: {
+            stat_descriptors: [
+              {
+                key: "product-export-count",
+                name: "Product count to export",
+                message: `There will be ${productCount} products exported by this action`,
+              },
+            ],
+          },
+        })
+    })
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJob: CreateBatchJobInput,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    req: Express.Request
+  ): Promise<CreateBatchJobInput> {
+    const {
+      limit,
+      offset,
+      order,
+      fields,
+      expand,
+      filterable_fields,
+      ...context
+    } = (batchJob?.context ?? {}) as ProductExportBatchJobContext
+
+    const listConfig = prepareListQuery(
+      {
+        limit,
+        offset,
+        order,
+        fields,
+        expand,
+      },
+      {
+        isList: true,
+        defaultRelations: this.defaultRelations_,
+      }
+    )
+
+    batchJob.context = {
+      ...(context ?? {}),
+      list_config: listConfig,
+      filterable_fields,
+    }
+
+    return batchJob
+  }
+
+  async processJob(batchJobId: string): Promise<void> {
+    let offset = 0
+    let limit = this.BATCH_SIZE
+    let advancementCount = 0
+    let productCount = 0
+
+    return await this.atomicPhase_(
+      async (transactionManager) => {
+        let batchJob = (await this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)) as ProductExportBatchJob
+
+        const { writeStream, fileKey, promise } = await this.fileService_
+          .withTransaction(transactionManager)
+          .getUploadStreamDescriptor({
+            name: "product-export",
+            ext: "csv",
+          })
+
+        const header = await this.buildHeader(batchJob)
+        writeStream.write(header)
+
+        advancementCount =
+          batchJob.result?.advancement_count ?? advancementCount
+        offset = (batchJob.context?.list_config?.skip ?? 0) + advancementCount
+        limit = batchJob.context?.list_config?.take ?? this.BATCH_SIZE
```

Comment:
We should consider what the limit means in terms of batch jobs vs. in terms of listing products and pagination. 

Limit could be interpreted as the total number of records to export and that by including it you export a "page" from the pagination, meaning the limit is interpreted in the same way it is in the ui. In that case we need to ensure that the productCount we are iterating towards is `offset + limit`. A use case for interpreting limit this way, could be something along the lines of "I want the 10 newest products". We could include a separate parameter for BATCH_SIZE, but given a predefined size between 50-100 it shouldn't matter that much.

If we choose to interpret "limit" as batch_size we should consider if we need to support limiting export size in a different way.

Reply:
I see your point, in that case we should keep limit/offset for pagination and probably add a new parameter which would be take as a number and if provided it means that we take only that number of products?

That way, if take is provided, we do not loop for the pagination and only take that number from the offset, otherwise it is everything and it is paginate. wdyt?

Reply:
also, since that job can't have a retry strategy, I could simplify that part of the process :)

Reply:
I think so, we should keep the terminology of the pagination parameters, I think just calling it batch_size in the context, however I think it might be better to merge the pr and create a new if that is the only point missing 😄

Reply:
😂 limit/offset represents the pagination and will not change, we just need to find the right wording for the new one. Batch size on the class represent the default limit at the moment, maybe it should be renamed that way too

Reply:
yeah I know, my point was that if limit represents the page size, that it maybe should represent the export length such that you can export the view from admin, and then include a new parameter for the size of each batch that is being uploaded 😄 

wdyt of merging and creating a new pr?

Reply:
Yeah agreed on the merge, just need to push the last changes